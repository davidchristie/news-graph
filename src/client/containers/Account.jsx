import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { browserHistory, Link } from 'react-router'

import { login, logout } from '../actions/account'
import Login from '../components/Login'
import Logout from '../components/Logout'

const Account = class Account extends Component {
  render () {
    const dispatch = this.props.dispatch
    let content = null
    if (this.props.profile) {
      const clickLogout = () => {
        dispatch(logout())
        browserHistory.push('/')
      }
      content = (
        <ul className='nav navbar-nav navbar-right'>
          <li className='nav-link' onClick={clickLogout}>
            <Logout onClick={clickLogout} />
          </li>
        </ul>
      )
    } else {
      const submitLogin = event => {
        event.preventDefault()
        const form = event.target
        const inputs = form.getElementsByTagName('input')
        const email = inputs[0].value
        const password = inputs[1].value
        dispatch(login({email, password}))
      }
      content = (
        <ul className='nav navbar-nav navbar-right'>
          <li className='nav-link'>
            <Link to='/join' className='btn btn-primary'>Signup</Link>
          </li>
          <li className='nav-link'>
            <Login onSubmit={submitLogin} />
          </li>
        </ul>
      )
    }
    return content
  }
}

Account.propTypes = {
  profile: PropTypes.shape({
    name: PropTypes.string.isRequired
  })
}

export default connect(state => {
  return {
    profile: state.app.account.profile
  }
})(Account)
