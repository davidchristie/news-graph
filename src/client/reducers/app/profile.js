// Profile:
// {
//   name
//   token
// }

export default (state = null, action) => {
  switch (action.type) {
    case 'LOGIN_SUCCESS':
      return action.profile
    case 'LOGOUT':
      return null
    case 'SIGNUP_SUCCESS':
      return action.profile
    default:
      return state
  }
}
