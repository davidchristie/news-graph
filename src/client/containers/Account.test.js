/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Account } from './Account'

import profile from '../examples/profile'

test('Account renders correctly', () => {
  const wrapper = shallow(<Account profile={profile} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
