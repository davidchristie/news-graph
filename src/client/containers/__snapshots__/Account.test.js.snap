exports[`test Account renders correctly 1`] = `
<ul
  className="nav navbar-nav navbar-right">
  <li
    className="nav-link"
    onClick={[Function]}>
    <Logout
      onClick={[Function]} />
  </li>
</ul>
`;
